migration-engine/connectors/migration-connector/src/migration_persistence.rs

```diff
@@ -99,17 +106,49 @@ pub trait IsWatchMigration {
     fn is_watch_migration(&self) -> bool;
 }
 
+pub struct NewMigration {
+    pub name: String,
+    pub datamodel_string: String,
+    pub datamodel_steps: Vec<MigrationStep>,
+    pub database_migration: serde_json::Value,
+}
+
 impl Migration {
-    pub fn new(name: String) -> Migration {
+    pub fn new(params: NewMigration) -> Migration {
+        let NewMigration {
+            name,
+            datamodel_string,
+            datamodel_steps,
+            database_migration,
+        } = params;
+
         Migration {
-            name: name,
+            name,
             revision: 0,
             status: MigrationStatus::Pending,
-            datamodel_string: String::new(),
+            datamodel_string,
+            datamodel_steps,
             applied: 0,
             rolled_back: 0,
+            database_migration,
+            errors: Vec::new(),
+            started_at: Self::timestamp_without_nanos(),
+            finished_at: None,
+        }
+    }
+
+    /// This is only useful for tests. Use `Migration::new()` if you want to initialize a valid
+    /// migration.
+    pub fn empty(name: String) -> Migration {
```

Comment:
maybe add `#[cfg(test)]` on top of this?

Reply:
It's used in other crates (`migration-engine-tests`), so I can't, unfortunately.